x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/es_index/es_index_params.tsx

```diff
@@ -20,38 +34,153 @@ export const IndexParamsFields = ({
+  );
+  const [usePreconfiguredSchema, setUsePreconfiguredSchema] = useState<boolean>(false);
+
+  useEffect(() => {
+    setDocumentToIndex(getDocumentToIndex(documents));
+  }, [documents]);
+
+  useEffect(() => {
+    if (actionConnector?.id === AlertHistoryEsIndexConnectorId) {
+      setUsePreconfiguredSchema(true);
+      editAction('documents', [JSON.stringify(AlertHistoryDocumentSchema)], index);
+      setDocumentToIndex(JSON.stringify(AlertHistoryDocumentSchema));
+    } else {
+      setUsePreconfiguredSchema(false);
+      editAction('documents', undefined, index);
+      setDocumentToIndex(undefined);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [actionConnector?.id]);
 
   const onDocumentsChange = (updatedDocuments: string) => {
     try {
       const documentsJSON = JSON.parse(updatedDocuments);
       editAction('documents', [documentsJSON], index);
+      setDocumentToIndex(updatedDocuments);
     } catch (e) {
       // set document as empty to turn on the validation for non empty valid JSON object
       editAction('documents', [{}], index);
+      setDocumentToIndex(undefined);
     }
   };
 
-  return (
+  const documentsFieldLabel = i18n.translate(
+    'xpack.triggersActionsUI.components.builtinActionTypes.indexAction.documentsFieldLabel',
+    {
+      defaultMessage: 'Document to index',
+    }
+  );
+
+  const resetDefaultIndex =
+    indexOverride && indexOverride !== AlertHistoryDefaultIndexName ? (
+      <EuiText size="xs">
+        <EuiLink
+          data-test-subj="resetDefaultIndex"
+          onClick={() => {
+            editAction('indexOverride', AlertHistoryDefaultIndexName, index);
+            setAlertHistoryIndexSuffix(defaultAlertHistoryIndexSuffix);
+          }}
+        >
+          <EuiIcon type="refresh" />
+          <FormattedMessage
+            id="xpack.triggersActionsUI.sections.alertsList.resetDefaultIndexLabel"
+            defaultMessage="Reset Default Index"
+          />
+        </EuiLink>
+      </EuiText>
+    ) : (
+      <></>
+    );
+
+  const preconfiguredDocumentSchema = (
+    <>
+      <EuiFormRow
+        fullWidth
+        error={errors.indexOverride as string[]}
+        isInvalid={(errors.indexOverride as string[]) && errors.indexOverride.length > 0}
+        label={i18n.translate(
+          'xpack.triggersActionsUI.components.builtinActionTypes.indexAction.preconfiguredIndex',
+          {
+            defaultMessage: 'ES Index',
+          }
+        )}
+        labelAppend={resetDefaultIndex}
+        helpText={
+          <>
+            <FormattedMessage
+              id="xpack.triggersActionsUI.components.builtinActionTypes.indexAction.preconfiguredIndexHelpText"
+              defaultMessage="Documents will be indexed into the {alertHistoryIndex} index"
+              values={{ alertHistoryIndex: `${ALERT_HISTORY_PREFIX}${alertHistoryIndexSuffix}` }}
+            />
+            <EuiSpacer size="xs" />
+            <EuiLink
```

Comment:
I think we can get this to be a shorter link (and in which case hopefully just tack it onto the same line as the help text above without the spacer between). 

Perhaps `View documentation` is enough? Or `View alertHistoryIndex documentation`, but I feel like the first might be enough.  I don't believe we recommend 'Learn more' links (even with the extra description).

Pinging @gchaps to double-check me :)

Reply:
Updated!

Reply:
Would "View docs" fit on the same line as the description?

Reply:
Updated to "View docs" in https://github.com/elastic/kibana/pull/94909/commits/f3fea184ebc1e3e02f57b085216f9eab918bae29